### ISSUE TITLE
Write UI tests for UI integration code (back handling, etc) and samples using AndroidX Test

### ISSUE THREAD

**Issue by zach-klippenstein** [square/workflow#615]
Announced at Google I/O 2018, AndroidX Test lets you write UI tests that you can run as unit tests.

Article from Android Developers here: https://medium.com/androiddevelopers/write-once-run-everywhere-tests-on-android-88adb2ba20c5

Docs here: https://developer.android.com/training/testing/

**Comment by rjrjr** [comment]
We can unit test the bits of it, but I don't see how we can get an integration test out of this.

**Comment by zach-klippenstein** [comment]
Why not? We can write integration tests for our sample apps:

```kotlin
@RunWith(AndroidJUnit4::class)
class LoginActivityTest {
  @Test fun successfulLogin() {
    // GIVEN
    val scenario = 
        ActivityScenario.launch(LoginActivity::class.java)
    // WHEN
    onView(withId(R.id.user_name)).perform(typeText(“test_user”))
    onView(withId(R.id.password))
        .perform(typeText(“correct_password”))
    onView(withId(R.id.button)).perform(click())
    // THEN
    assertThat(getIntents().first())
        .hasComponentClass(HomeActivity::class.java)
 }
}
```

**Comment by rjrjr** [comment]
Whoa! I hadn't read https://medium.com/androiddevelopers/write-once-run-everywhere-tests-on-android-88adb2ba20c5 when I posted that. I had no idea how ambitious that thing is.

**Comment by rjrjr** [comment]
Closed by #618
